### PR TITLE
neovim: update to 8.0.0

### PR DESCRIPTION
**Pull request by Grimler91**
Updated package starts without issues, but I am not a (neo)vim user<sup><sub>don't trust agnostic-apollo's contradictory claims</sub></sup> so haven't tested it more thoroughly. Would be great if someone who actually uses neovim could test the debs

Fixes https://github.com/termux/termux-packages/issues/12226.

### CODE REVIEW

File: packages/neovim/runtime-autoload-man.vim.patch [hunk 1]
```diff
@@ -1,12 +0,0 @@
-diff -u -r ../neovim-0851057a8deaa1197bd0af22babb62c1146d836c/runtime/autoload/man.vim ./runtime/autoload/man.vim
---- ../neovim-0851057a8deaa1197bd0af22babb62c1146d836c/runtime/autoload/man.vim	2018-02-04 12:54:30.000000000 +0000
```
Comment (licy183) [inline packages/neovim/runtime-autoload-man.vim.patch:2]: This `.vim` file has been rewritten upstream using a lua script. So we should patch this script. See 
https://github.com/termux/termux-packages/blob/588653cc210ff6a63fc429cd7609d349aa82bdee/packages/neovim-nightly/runtime-lua-man.lua.patch.

File: packages/neovim-nightly/build.sh [hunk 2]
```diff
@@ -6,6 +6,7 @@ TERMUX_PKG_MAINTAINER="Aditya Alok <alok@termux.org>"
 # "0.8.0-dev-698-g1ef84547a" < "0.8.0-dev-nightly-10-g1a07044c1", we need to bump
 # the epoch of the package version.
 TERMUX_PKG_VERSION="1:0.8.0-dev-1202-g33dd917d7"
+TERMUX_PKG_REVISION=1
```
Comment (licy183) [inline packages/neovim-nightly/build.sh:9]: I added a patch to revert the change that `neovim` should use `libvterm=0.3` yesterday. As for now, this patch is useless, I think we should remove it.

File: packages/neovim/build.sh [hunk 3]
```diff
@@ -40,13 +38,6 @@ termux_step_host_build() {
 	chmod +x "$TERMUX_ORIGINAL_CMAKE"
 	export TERMUX_ORIGINAL_CMAKE="$TERMUX_ORIGINAL_CMAKE.orig"
 
-	mkdir -p $TERMUX_PKG_HOSTBUILD_DIR/deps
-	cd $TERMUX_PKG_HOSTBUILD_DIR/deps
-	cmake $TERMUX_PKG_SRCDIR/third-party
```
Comment (licy183) [inline packages/neovim/build.sh:45]: I think this should not be removed. See https://github.com/termux/termux-packages/commit/2c77b5406c9b6e8ff056f266544df336439ca6e5.

**Comment by Terprogram** [comment]
Good

**Comment by CanKolay3499** [comment]
Neovim nightly works fine. so, there's no reasons to not merge this.

**Comment by thunder-coding** [comment]
> Neovim nightly works fine. so, there's no reasons to not merge this.

We are a rolling distribution, so this means that we'll have to catch up with the latest version today or in a while. Better it be today than late

**Review by thunder-coding — APPROVED** [review]
As a neovim user, I can confirm that all my installed plugins and config work pretty well with Neovim v0.8. No problems so far, lsp's also working as usual great!
